deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/AgentJanitor.java

```diff
@@ -68,6 +72,10 @@ private Set<String> getTerminatedHostsFromSource(List<String> staleHostIds) {
                         .getTerminatedHosts(staleHostIds.subList(i, Math.min(i + batchSize, staleHostIds.size()))));
             } catch (Exception ex) {
                 LOG.error("Failed to get terminated hosts", ex);
+
+                Metrics.counter(CUSTOM_NAME_PREFIX + "error-budget.counters",
+                        "response_type", "failure",
+                        "method_name", this.getClass().getSimpleName()).increment();
```

Comment:
Since the counter doesn't change, it's better to store the counter in a local private field. You can initialize it in the constructor. See https://micrometer.io/docs/concepts#_global_registry

Reply:
Applied on worker: SimpleAgentJanitor.java
Let me know if it's OK, so I can replicate it to all of them.

https://github.com/pinterest/teletraan/pull/1286/files#diff-d325e9aa7284f0411829cb75934a0ddae12c7281fda3b09da7779c406e8bad3cR69

Reply:
Looks good